server/webhook/merge_request.go

```diff
@@ -29,57 +29,140 @@ func (w *webhook) handleDMMergeRequest(event *gitlab.MergeEvent) ([]*HandleWebho
+					handlers = append(handlers, &HandleWebhook{
+						Message: message,
+						ToUsers: updatedCurrentUsers,
+						From:    senderGitlabUsername,
+					})
 				}
 			}
 
-			message = fmt.Sprintf("[%s](%s) assigned you to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+			// Handle change in reviewers
+			if event.Changes.Reviewers.Current != nil || event.Changes.Reviewers.Previous != nil {
+				isAuthorPresent, updatedPreviousUsers, updatedCurrentUsers := w.handleDMNotificationUsers(event.ObjectAttributes.AuthorID, event.Changes.Reviewers.Previous, event.Changes.Reviewers.Current)
+
+				// Check if the MR author is present in the assignee changes
+				if !isAuthorPresent {
+					message = fmt.Sprintf("[%s](%s) updated the list of reviewers for the merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+					handlers = append(handlers, &HandleWebhook{
+						Message: message,
+						ToUsers: []string{authorGitlabUsername},
+						From:    senderGitlabUsername,
+					})
+				}
+
+				if len(updatedPreviousUsers) != 0 {
+					message = fmt.Sprintf("[%s](%s) removed you as a reviewer from the merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+					handlers = append(handlers, &HandleWebhook{
+						Message: message,
+						ToUsers: updatedPreviousUsers,
+						From:    senderGitlabUsername,
+					})
+				}
+
+				if len(updatedCurrentUsers) != 0 {
+					message = fmt.Sprintf("[%s](%s) assigned you as a reviewer to merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+					handlers = append(handlers, &HandleWebhook{
+						Message: message,
+						ToUsers: updatedCurrentUsers,
+						From:    senderGitlabUsername,
+					})
+				}
+			}
+
+			// Check if multiple events happened in single webhook event
+			if w.checkForMultipleEvents(event) {
+				message = fmt.Sprintf("[%s](%s) updated the merge request [%s!%v](%s)", senderGitlabUsername, w.gitlabRetreiver.GetUserURL(senderGitlabUsername), event.ObjectAttributes.Target.PathWithNamespace, event.ObjectAttributes.IID, event.ObjectAttributes.URL)
+				handlers = []*HandleWebhook{{
+					Message: message,
+					ToUsers: toUsers,
+					From:    senderGitlabUsername,
+				}}
```

Comment:
Should we append here instead of reassigning `handlers` to an array with one element? I would still want to receive the notification for being assigned to an issue/MR

Reply:
We have not appended here as there are multiple events. If we just append it to the array, then there will be multiple posts on the mattermost channel for the single event. Shouldn't we just post a single time for an event? Please let me know your opinions on this.